### PR TITLE
Fix: Off by 1 Error in Old Sha

### PR DESCRIPTION
**Pull request by harrisoncramer**

### CODE REVIEW

**Comment by harrisoncramer** [comment]
Fixes #193.

@jakubbortlik Would you please confirm this fixes your problem?

File: lua/gitlab/hunks/init.lua [hunk 1]
```diff
@@ -35,28 +35,31 @@ end
...
+---@return boolean
 local line_was_removed = function(linnr, hunk, all_diff_output)
   for matching_line_index, line in ipairs(all_diff_output) do
     local found_hunk = M.parse_possible_hunk_headers(line)
     if found_hunk ~= nil and vim.deep_equal(found_hunk, hunk) then
       -- We found a matching hunk, now we need to iterate over the lines from the raw diff output
       -- at that hunk until we reach the line we are looking for. When the indexes match we check
       -- to see if that line is deleted or not.
-      for hunk_line_index = found_hunk.old_line, hunk.old_line + hunk.old_range - 1, 1 do
+      for hunk_line_index = found_hunk.old_line, hunk.old_line + hunk.old_range, 1 do
```
Comment (harrisoncramer) [inline lua/gitlab/hunks/init.lua:46]: This was the actual bug I believe...

File: lua/gitlab/hunks/init.lua [hunk 2]
```diff
@@ -173,63 +177,58 @@ local count_changes = function(lines)
...
+  end
+
+  return List.new(hunks):find(function(hunk)
+    local old_line_end = hunk.old_line + hunk.old_range
+    local new_line_end = hunk.new_line + hunk.new_range
+    local in_old_range = old_line >= hunk.old_line and old_line <= old_line_end
+    local in_new_range = old_line >= hunk.new_line and new_line <= new_line_end
+    return (in_old_range or in_new_range) and line_was_removed(old_line, hunk, all_diff_output)
+  end) and "deleted" or "unmodified"
+end
```
Comment (harrisoncramer) [inline lua/gitlab/hunks/init.lua:213]: Using the List class makes this code much easier to understand IMO

**Comment by jakubbortlik** [comment]
> Fixes #193.
> 
> @jakubbortlik Would you please confirm this fixes your problem?

Hi, yes, in the `fix-index-issue` branch the problem is gone. Thanks!

**Pull request merged**